File: csharp/BronKerbosch/BronKerbosch2bGPX.cs
// Bron-Kerbosch algorithm with pivot of highest degree towards the remaining candidates (IK_GPX)

using BronKerbosch;
using System.Collections.Generic;

internal static class BronKerbosch2bGPX<VertexSet, VertexSetMgr>
    where VertexSet : ISet<Vertex>
    where VertexSetMgr : IVertexSetMgr<VertexSet>
{
#   pragma warning disable IDE0022 // Use expression body for method
    public static void Explore(UndirectedGraph<VertexSet, VertexSetMgr> graph, ICliqueConsumer consumer)
    {
        Pivot<VertexSet, VertexSetMgr>.Explore(graph, consumer, PivotChoice.MaxDegreeLocalX);
    }
}

File: rust/bron_kerbosch/src/core/bron_kerbosch2b_gpx.rs
//! Bron-Kerbosch algorithm with pivot of highest degree towards the remaining candidates (IK_GPX)

use super::algorithm::BronKerboschAlgorithm;
use super::bron_kerbosch_pivot::{PivotChoice, explore_with_pivot};
use super::clique_consumer::CliqueConsumer;
use super::graph::Graph;
use super::vertexsetlike::VertexSetLike;

pub struct Algo();
impl BronKerboschAlgorithm for Algo {
    fn name() -> String {
        String::from("Ver2½-GPX")
    }

    fn explore<VertexSet, Consumer>(
        graph: &Graph<VertexSet>,
        consumer: Consumer,
    ) -> Consumer::Harvest
    where
        VertexSet: VertexSetLike,
        Consumer: CliqueConsumer,
    {
        explore_with_pivot(graph, consumer, PivotChoice::MaxDegreeLocalX)
    }
}
